### PR TITLE
add MongoDB plugin

### PR DESCRIPTION
**Pull request by jipperinbham**
This PR contains a few additional changes along with a MongoDB plugin:

- `testutil.Accumulator` was using a `Value` and `Values` field and was using a specific one depending on whether the operation was `Add` vs `AddValuesWithTime`, this was causing tests that used `AddValuesWithTime` to not function properly so I removed `Value` and reworked how the `Add` function worked to be more inline with the actual `Accumulator` implementation.

- I copied over [stat_types.go](https://github.com/mongodb/mongo-tools/blob/master/mongostat/stat_types.go) from MongoDB Inc's [mongo-tools](https://github.com/mongodb/mongo-tools) project but removed any code that was requiring another dependency in that project as well as removed any unused code. That project is licensed under Apache V2 so I made note of the origination of work and any modifications in the file.

### CODE REVIEW

File: plugins/mongodb/mongodb.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		} else if u.Scheme == "" {
+			u.Scheme = "mongodb"
+			// fallback to simple string based address (i.e. "10.0.0.1:10000")
+			u.Host = serv
+			if u.Path == u.Host {
+				u.Path = ""
+			}
+		}
+		wg.Add(1)
+		go func(serv string) {
```
Comment (otoolep) [inline plugins/mongodb/mongodb.go]: Where is `serv` used in the function launched by the goroutine?

File: plugins/mongodb/mongodb.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	var outerr error
+
+	for _, serv := range m.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("Unable to parse to address '%s': %s", serv, err)
+		} else if u.Scheme == "" {
+			u.Scheme = "mongodb"
+			// fallback to simple string based address (i.e. "10.0.0.1:10000")
+			u.Host = serv
```
Comment (jipperinbham) [inline plugins/mongodb/mongodb.go]: right here

File: plugins/mongodb/mongodb.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		m.gatherServer(m.getMongoServer(localhost), acc)
+		return nil
+	}
+
+	var wg sync.WaitGroup
+
+	var outerr error
+
+	for _, serv := range m.Servers {
+		u, err := url.Parse(serv)
```
Comment (jipperinbham) [inline plugins/mongodb/mongodb.go]: and here

File: plugins/mongodb/mongodb.go (hunk 1 above)
Comment (jipperinbham) [inline plugins/mongodb/mongodb.go]: Good catch, it's not needed for the go routine

**Comment by jipperinbham** [comment]
@otoolep I removed the unused `serv` variable as well as updated the sample config.toml and README

**Comment by otoolep** [comment]
Great, thanks.

**Pull request merged**